nomad/structs/config/vault.go

```diff
@@ -0,0 +1,131 @@
+package config
+
+import vault "github.com/hashicorp/vault/api"
+
+// VaultConfig contains the configuration information necessary to
+// communicate with Vault in order to:
+//
+// - Renew Vault tokens/leases.
+//
+// - Pass a token for the Nomad Server to derive sub-tokens.
+//
+// - Create child tokens with policy subsets of the Server's token.
+type VaultConfig struct {
+
+	// RoleName is the Vault role in which Nomad will derive child tokens using
+	// /auth/token/create/[role_name]
+	RoleName string `mapstructure:"role_name"`
+
+	// RoleToken is the periodic Vault token given to Nomad such that it can
+	// derive child tokens. The RoleToken should be created from the passed
+	// RoleName. Nomad will renew this token at half its lease lifetime.
+	RoleToken string `mapstructure:"role_token"`
```

Comment:
Can we rename this to just `Token` (or a PeriodicToken, if the token is *always* periodic)?